www/js/metrics-factory.js

```diff
@@ -0,0 +1,157 @@
+'use strict';
+
+angular.module('emission.main.metrics.factory', [])
+
+.factory('FootprintHelper', function() {
+  var fh = {};
+  var footprint = {
+    train: 92/1609,
+    car: 287/1609,
+    ON_FOOT: 0,
+    BICYCLING: 0
+  }
+  var readable = function(v) {
+    return v > 9999? Math.round(v / 1000) + 'k kg CO₂' : Math.round(v) + ' kg CO₂';
+  }
+  var mtokm = function(v) {
+    return v / 1000;
+  }
+  fh.getFootprintRaw = function(distance, mode) {
+    if (mode === "IN_VEHICLE") {
+      return [footprint.train * mtokm(distance), footprint.car * mtokm(distance)];
+    } else {
+      return footprint[mode] * mtokm(distance);
+    }
+  }
+  fh.getFootprint = function(distance, mode) {
+    if (mode === "IN_VEHICLE") {
+      return readable(footprint.train * mtokm(distance)) + ' ~ ' + readable(footprint.car * mtokm(distance));
+    } else {
+      return readable(footprint[mode] * mtokm(distance));
+    }
+  }
+  return fh;
+})
+
+.factory('CalorieHelper', function(){
+  Number.prototype.between = function (min, max) {
```

Comment:
nice! I've generally been putting load/store functions into the factory instead of having a separate module for load/store like you do with the user data.